pkg/charts/cni/values.go

```diff
@@ -17,22 +17,27 @@ const (
 
 // Values contains the top-level elements in the cni Helm chart
 type Values struct {
-	InboundProxyPort    uint   `json:"inboundProxyPort"`
-	OutboundProxyPort   uint   `json:"outboundProxyPort"`
-	IgnoreInboundPorts  string `json:"ignoreInboundPorts"`
-	IgnoreOutboundPorts string `json:"ignoreOutboundPorts"`
-	CliVersion          string `json:"cliVersion"`
-	CNIPluginImage      string `json:"cniPluginImage"`
-	CNIPluginVersion    string `json:"cniPluginVersion"`
-	LogLevel            string `json:"logLevel"`
-	PortsToRedirect     string `json:"portsToRedirect"`
-	ProxyUID            int64  `json:"proxyUID"`
-	DestCNINetDir       string `json:"destCNINetDir"`
-	DestCNIBinDir       string `json:"destCNIBinDir"`
-	UseWaitFlag         bool   `json:"useWaitFlag"`
-	PriorityClassName   string `json:"priorityClassName"`
-	ProxyAdminPort      string `json:"proxyAdminPort"`
-	ProxyControlPort    string `json:"proxyControlPort"`
+	InboundProxyPort    uint         `json:"inboundProxyPort"`
+	OutboundProxyPort   uint         `json:"outboundProxyPort"`
+	IgnoreInboundPorts  string       `json:"ignoreInboundPorts"`
+	IgnoreOutboundPorts string       `json:"ignoreOutboundPorts"`
+	CliVersion          string       `json:"cliVersion"`
+	CNIPluginImage      string       `json:"cniPluginImage"`
+	CNIPluginVersion    string       `json:"cniPluginVersion"`
+	LogLevel            string       `json:"logLevel"`
+	PortsToRedirect     string       `json:"portsToRedirect"`
+	ProxyUID            int64        `json:"proxyUID"`
+	DestCNINetDir       string       `json:"destCNINetDir"`
+	DestCNIBinDir       string       `json:"destCNIBinDir"`
+	UseWaitFlag         bool         `json:"useWaitFlag"`
+	PriorityClassName   string       `json:"priorityClassName"`
+	ProxyAdminPort      string       `json:"proxyAdminPort"`
+	ProxyControlPort    string       `json:"proxyControlPort"`
+	Tolerations         []Toleration `json:"tolerations"`
```

Comment:
I think this needs to be a more generic `interface{}` to allow users to set tolerations which use more than just the `Operator` field.  Or add all of the possible fields to the `Toleration` struct.

Reply:
Thank you. Matei said something similar but I didn't understand the implication for users overriding the setting.